content/en/docs/memory-safety.md

```diff
@@ -0,0 +1,120 @@
+
+If we take for granted that using an unsafe language will produce some number of vulnerabilities, the question we'd want to ask is: are there techniques we can undertake to reduce this risk, without forcing ourselves to entirely change programming languages? And the answer is absolutely yes. Not all projects written in unsafe languages are equally unsafe and unreliable.
+
+Some practices which can lower the risk of using an unsafe language are:
+
+- Using [some modern C++ idioms] which can help produce more safe and reliable
+  code
+- Using [fuzzers] and [sanitizers] to help find bugs before they make it into
+  production
+- Using exploit mitigations to help increase the difficulty of exploiting
+  vulnerabilities
+- Privilege separation so that even when a vulnerability is exploited, the blast radius is smaller
+
+These practices meaningfully lower the risk of using an unsafe language, and if we've failed to convince you to change languages, and you are going to continue to write C and C++, adopting these is an imperative. Unfortunately, they are also woefully insufficient.
+
+The people who are the forefront of developing modern C++ idioms, fuzzers, sanitizers, exploit mitigations, and privilege separation techniques are browser and operating system developers — precisely the groups we highlighted at the start with statistics about the prevalence of memory safety problems. Despite these teams' investment in these techniques, their use of unsafe languages weighs them down. At pwn2own, a large hacking competition, in 2019 [over half of vulnerabilities exploited] in these products were due to a lack of memory safety, and with one exception, [every successful attack exploited] at least one memory safety vulnerability.
+
+## Is dropping C and C++ really practical?
+
+Hopefully by now we've convinced you that unsafe languages like C and C++ are fundamental root causes of huge swathes of the insecurity in our products, and that while there are practices you can undertake to reduce the risk, you can't get anywhere close to eliminating it. All of which may still leave you with a feeling that changing the programming language you use, to produce millions of lines of code, is an overwhelmingly large task. By breaking it down into manageable pieces, we can start making progress — our goal is not one big-bang rewrite-the-world, but rather to make progress towards reducing our risk.
+
+The first place to start is with brand new projects. For these, you have the choice to simply use a memory safe language. These have the lowest risk, because you do not need to start by rewriting any code, though projects like this often do require improvements to testing or deployment infrastructure to support a new programming language. This was the approach taken in ChromeOS's CrosVM, a brand new component of the operating system.
```

Comment:
Another example possibly worth mentioning: firecracker from AWS.

Reply:
Thanks! I'm going to merge this with the typo fixes then make more editorial changes later.